libstorage/drivers/integration/linux/linux.go

```diff
@@ -273,13 +275,28 @@ func (d *driver) Mount(
 		return "", nil, goof.New("no device name returned")
 	}
 
+	mountPath, err := d.getVolumeMountPath(vol.Name)
+	if err != nil {
+		return "", nil, err
+	}
+
 	mounts, err := client.OS().Mounts(
 		ctx, ma.DeviceName, "", opts.Opts)
 	if err != nil {
 		return "", nil, err
 	}
 
 	if len(mounts) > 0 {
+		if _, ok := ctx.Value(ctxExactMountKey).(interface{}); ok {
+			for _, m := range mounts {
+				if m.MountPoint == mountPath {
+					ctx.Debug("returning existing mount")
+					return d.volumeMountPath(
+						m.MountPoint), vol, nil
+				}
+			}
+			return "", nil, goof.New("device is already mounted")
+		}
```

Comment:
This code previously just returned the first mount it saw with the matching device name. That's not good enough when we are using bind mounts, as there will be multiple entries. That's why the "exactmatch" context was added -- we only return the mountpath for a mount matching both the device and the expected mountpoint.

For example, if the mount table had this:
```
/dev/xvdf /mnt/test xfs rw,seclabel,relatime,nouuid,attr2,inode64,noquota 0 0
/dev/xvdf /var/lib/rexray/volumes/test/data xfs rw,seclabel,relatime,nouuid,attr2,inode64,noquota 0 0
```
We would return `/mnt/test/data` for the mount path, when we really wanted `/var/lib/rexray/volumes/test/data`